civihr_employee_portal/js/ta-documents-app.js

```diff
@@ -46,17 +44,14 @@
          * @param {string} mode
          */
         function modalDocument (id, role, mode) {
-          $rootScope.$broadcast('ct-spinner-show', 'document-' + id);
-          vm.loadingModalData = true;
-
           DocumentService.get({ id: id })
             .then(function(data) {
               if (!data) {
                 throw new Error('Requested Document is not available');
               }
-
+              vm.openModalDocument(data[0], role, mode);
```

Comment:
why is `openModal` published on the `vm` object? Here you are just calling it internally. Is the function called also on the view?

Reply:
You mean `openModalDocument()`? No, its not used any where else. I will make it private.